### PR TITLE
Pickers: add in the ability to have controlled pickers.

### LINKED ISSUE #928: Reset selected items state of TagPicker
I am in a situation where I have a reset button to reset the values of a `TagPicker`. 

Since the component is alrready mounted, the internal state is not reset if I change the `defaultSelectedItems` property on the `TagPicker`. 

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx#L51

It would be cool if was possible to reset the selected items state on the component. One solution I can think of right now would be to change the `defaultSelectedItems` property and the call `tagPicker.forceUpdate()`, but it does not seem so elegant. 

Thanks.

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #928 and #1007 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Pickers: add in the ability to have controlled pickers.

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by joschect** [comment]
@MLoughry Definitely, but if you don't mind me asking, which part is particularly confusing? It could be that the confusing logic indicates something may be done poorly and I'd like to correct that if possible. I'm too close to most of this picker stuff so input is greatly appreciated.

**Comment by MLoughry** [comment]
It's just a lot of interlocking functions that you have to jump around to read. I'm just used to OWA code at this point, where (with a few exceptions) we try to keep files under 100-200 lines or so and break them into sub-components or external actions to keep the individual components simple.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.Props.ts [hunk 1]
```diff
@@ -66,6 +66,9 @@ export interface IBasePickerProps<T> extends React.Props<any> {
    * @default false
    */
   disabled?: boolean;
+
+
+  selectedItems?: T[];
```
Comment (kristoferbrown) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.Props.ts]: Add a comment explaining what this prop does.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.Props.ts]: @joschect ping

Comment (devyhia) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.Props.ts]: @joschect Any updates on this one? I am blocked by this feature.

**Comment by dzearing** [comment]
Closing per John's request.

**Comment by devyhia** [comment]
Is that change already merged?

**Pull request closed without merging**